ZuluContent/Engines/Craft/DefCarpentry.cs

```diff
@@ -1,255 +1,258 @@
+        {
+            get { return 1044004; } // <CENTER>CARPENTRY MENU</CENTER>
+        }
+
+        private static CraftSystem m_CraftSystem;
+
+        public static CraftSystem CraftSystem
+        {
+            get
+            {
+                if (m_CraftSystem == null)
+                    m_CraftSystem = new DefCarpentry();
+
+                return m_CraftSystem;
+            }
+        }
+
+        public override double GetChanceAtMin(CraftItem item)
+        {
+            return 0.0;
+        }
+
+        private DefCarpentry() : base(3, 3, 2)
+        {
+        }
+
+        public override int GetCraftSkillRequired(int itemSkillRequired, Type craftResourceType)
+        {
+            var resource = CraftResources.GetFromType(craftResourceType);
+            return itemSkillRequired + (int) (CraftResources.GetCraftSkillRequired(resource) / 4);
+        }
+
+        public override int GetCraftPoints(int itemSkillRequired, int materialAmount)
+        {
+            return (itemSkillRequired + materialAmount) * 10;
+        }
+
+        public override int CanCraft(Mobile from, BaseTool tool, Type itemType)
+        {
+            if (tool == null || tool.Deleted || tool.UsesRemaining < 0)
+                return 1044038; // You have worn out your tool!
+            else if (!BaseTool.CheckAccessible(tool, from))
+                return 1044263; // The tool must be on your person to use.
+
+            return 0;
+        }
+
+        public override void PlayCraftEffect(Mobile from)
+        {
+            from.PlaySound(0x23D);
+        }
+
+        public override int PlayEndingEffect(Mobile from, bool failed, bool lostMaterial, bool toolBroken, int quality,
+            bool makersMark, CraftItem item)
+        {
+            if (toolBroken)
+                from.SendLocalizedMessage(1044038); // You have worn out your tool
+
+            if (failed)
+            {
+                if (lostMaterial)
+                    return 1044043; // You failed to create the item, and some of your materials are lost.
+                else
+                    return 1044157; // You failed to create the item, but no materials were lost.
+            }
+            else
+            {
+                if (quality == 0)
+                    return 502785; // You were barely able to make this item.  It's quality is below average.
+                else if (makersMark && quality == 2)
+                    return 1044156; // You create an exceptional quality item and affix your maker's mark.
+                else if (quality == 2)
+                    return 1044155; // You create an exceptional quality item.
+                else
+                    return 1044154; // You create the item.
+            }
+        }
+
+        public override void InitCraftList()
+        {
+            int index = -1;
+
+            AddCraft(typeof(BarrelStaves), 1044294, 1027857, 00.0, 25.0, typeof(Log), 1044041, 5, 1044351);
```

Comment:
Planning on turning these into configs?

Reply:
Haven't really touched them yet but they could be configs indeed